app/models/user.rb

```diff
@@ -42,6 +42,11 @@ def confirm_friend(user)
     friendship1 = inverse_friendships.find { |friendship| friendship.user == user }
     friendship1.confirmed = true
     friendship1.save
+    friendship2 = user.inverse_friendships.find { |friendship| friendship.user == self } ||
+                  Friendship.create(user_id: friendship1.friend_id,
+                                    friend_id: friendship1.user_id)
+    friendship2.confirmed = true
+    friendship2.save
   end
```

Comment:
[x] In the create action, we added the confirmed: true to this action as commanded but furthermore we add validation to the Friendship model to avoid the creation of friendships with nil values. as this is a boolean field we have to use `validates :confirmed, exclusion: { in: [nil] }`

[x] For the `friends` method it was implemented the suggested change but we decided not to use the source attribute given that we thing the naming that we are using is self-explaining so intentionally avoid that change. however we are using the many to many relationship in all of those methods `friends_pending = friendships.select(:id, :friend_id).where(confirmed: false)`, **friends friend_requests pending_friends**

[x] the confirm_friend was re-implemented as suggested in the Friendship model with the predicted advantages, simpler syntax, one doesn't have to pass any variable to the method

[x] Additionally to all of these changes has been updated the testing because in principle everything gots broken